tests/test_databases.py

```diff
@@ -666,42 +683,47 @@ def run_sync(*args, **kwargs):
                 raw_connection = connection.raw_connection
 
                 # Insert query
-                if str(database_url).startswith("mysql"):
+                if database.url.scheme in ["mysql", "postgresql+aiopg"]:
                     insert_query = "INSERT INTO notes (text, completed) VALUES (%s, %s)"
                 else:
                     insert_query = "INSERT INTO notes (text, completed) VALUES ($1, $2)"
 
                 # execute()
                 values = ("example1", True)
 
-                if str(database_url).startswith("postgresql"):
-                    await raw_connection.execute(insert_query, *values)
-                elif str(database_url).startswith("mysql"):
+                if database.url.scheme in ["mysql", "postgresql+aiopg"]:
                     cursor = await raw_connection.cursor()
                     await cursor.execute(insert_query, values)
-                elif str(database_url).startswith("sqlite"):
+                elif database.url.scheme == "postgresql":
+                    await raw_connection.execute(insert_query, *values)
+                elif database.url.scheme == "sqlite":
                     await raw_connection.execute(insert_query, values)
 
                 # execute_many()
                 values = [("example2", False), ("example3", True)]
 
-                if str(database_url).startswith("mysql"):
+                if database.url.scheme == "mysql":
                     cursor = await raw_connection.cursor()
                     await cursor.executemany(insert_query, values)
+                elif database.url.scheme == "postgresql+aiopg":
+                    cursor = await raw_connection.cursor()
+                    # No async support for `executemany`
```

Comment:
Okay, in these kinds of cases we should still support the `.executemany` API, even if all it *actually* does is `for value in values: ...execute...` under the hood.

(We can document limitations of particular drivers, but it'd def. be a benefit if we can keep the API the same, even if there's different performance characteristics in some cases.)

Reply:
Yeah, I see and agree, will try my best.

Reply:
Ok, it's the test for the raw driver across different backends, but `aiopg` has the implementation for `execute_many` and yeah it's with the `for` loop.

Reply:
Grand, let's just hook our backend implementation up to that then, right?

Reply:
All should be in place! I have a unittest to test it for `database.execute_many`.

Reply:
this case is the battle test of the raw driver, so it replicates backend implementations in som places.